### ISSUE TITLE
Try to merge code references when using subtract

### ISSUE THREAD

**Issue by ocean90** [wp-cli/i18n-command#105]
Not sure how and if even possible, but would like to raise this request anyway.

The issue was [reported 6 years ago](https://core.trac.wordpress.org/ticket/20881) for makepot.php and is still valid for `i18n make-pot`.

**Example**
`wordpress.pot` includes the string "Edit" like this:

```
#: wp-includes/class-wp-editor.php:1262
#: wp-includes/class-walker-comment.php:236
#: wp-includes/class-walker-comment.php:340
msgid "Edit"
msgstr ""
```

Without subtracting `wordpress.pot`, `wordpress-admin.pot` also includes the string "Edit":

```
#: wp-admin/includes/class-wp-terms-list-table.php:437
#: wp-admin/includes/class-wp-users-list-table.php:410
#: wp-admin/includes/dashboard.php:634
#: wp-admin/includes/post.php:1356
#: wp-admin/includes/class-wp-ms-sites-list-table.php:513
#: wp-admin/includes/class-walker-nav-menu-edit.php:148
#: wp-admin/includes/meta-boxes.php:107
#: wp-admin/includes/meta-boxes.php:154
#: wp-admin/includes/meta-boxes.php:218
#: wp-admin/includes/class-wp-links-list-table.php:317
#: wp-admin/includes/class-wp-ms-users-list-table.php:335
#: wp-admin/includes/class-wp-ms-users-list-table.php:439
#: wp-admin/includes/class-wp-posts-list-table.php:396
#: wp-admin/includes/class-wp-posts-list-table.php:1238
#: wp-admin/includes/class-wp-comments-list-table.php:589
#: wp-admin/includes/class-wp-media-list-table.php:652
#: wp-admin/includes/class-wp-media-list-table.php:701
#: wp-admin/edit-form-comment.php:106
#: wp-admin/comment.php:216
msgid "Edit"
msgstr ""
```

Now when using `--substract=wordpress.pot`, `wordpress-admin.pot` would not contain the "Edit" string but `wordpress.pot` wasn't updated to include the "merged" code references.
I'm aware that it's not a merge so updating the files provided via substract might not something that someone would expect? Maybe do this only if another flag `--substract-and-merge` is provided?

**Comment by swissspidy** [comment]
As discussed privately with @ocean90, I think this is something that should be added as a separate command in the `wp i18n` namespace.

So what we're after is a `wordpress.pot` file that contains all source code references that otherwise would get lost because they get stripped out in `wordpress-admin.pot` because of the subtracting.

With separate commands it could work like this:

1. Run `wp i18n make-pot` to create a `wordpress.pot` file.
1. Run `wp i18n make-pot` to create a `wordpress-admin.pot` file (without any subtraction).
1. Run a command like `wp i18n add-references` to put all the code references together.  
  Example: `wp i18n add-references wordpress.pot wordpress-admin.pot` would update code references in `wordpress.pot` for all strings that also exist in `wordpress-admin.pot`.  
  In theory it could be made possible to pass more than one file to find references in
1. Now that you have `wordpress.pot` with all references, you need to subtract the strings from `wordpress-admin.pot`.  For this you could run a command like `wp i18n subtract-strings wordpress-admin.pot wordpress.pot`, potentially even passing more files than `wordpress.pot`

So tl;dr:

1. `wp i18n make-pot (…) wordpress.pot`
1. `wp i18n make-pot (…) wordpress-admin.pot`
1. `wp i18n make-pot (…) wordpress-network-admin.pot`
1. `wp i18n add-references wordpress.pot wordpress-admin.pot wordpress-network-admin.pot`
1. `wp i18n subtract-strings wordpress-admin.pot wordpress.pot wordpress-network-admin.pot`

**Comment by swissspidy** [comment]
Created a PR for this: #267
